src/fairness/print_hierarchy/flux_shares.cpp

```diff
@@ -0,0 +1,156 @@
+hierarchy of users. It will then parse each line, fetching tokens delimited by
+','. The formatting for lines will be different if a line contains a bank or if
+it contains an association.
+
+If parsable is false, the output of the database hierarchy will be
+formatted into columns.
+*/
+int print_hierarchy (const std::string &filename, bool parsable)
+{
+    std::stringstream buffer;
+    std::string line;
+
+    buffer = read_from_db (filename);
+    if (buffer.rdbuf ()->in_avail () == 0)
+        return -1;
+
+    if (parsable) {
+        std::cout << "Account|Username|RawShares|RawUsage" << std::endl;
+    } else {
+        std::cout << std::setw(20) << std::left << "Account"
+                  << std::setw(20) << std::right << "Username"
+                  << std::setw(20) << std::right << "RawShares"
+                  << std::setw(20) << std::right << "RawUsage"
+                  << std::endl;
+    }
+
+    while (std::getline (buffer, line)) {
+        std::string lvl, indent_lvl, acct, username, shrs, usg;
+        std::stringstream ss (line);
+
+        std::getline (ss, lvl, ',');
+        std::getline (ss, acct, ',');
+        std::getline (ss, username, ',');
+        std::getline (ss, shrs, ',');
+        std::getline (ss, usg, ',');
+
+        // determine indent based on level and add it to the account name
+        for (int i = 0; i < std::stoi (lvl); i++) {
+            indent_lvl += " ";
+        }
+        acct = indent_lvl + acct;
+
+        if (parsable)
+        {
+            if (username == "%^+_nouser")
+                std::cout << acct << "||" << shrs << "|" << usg << std::endl;
+            else
+                std::cout << acct << "|"
+                          << username << "|"
+                          << shrs << "|"
+                          << usg
+                          << std::endl;
+        } else {
+            if (username == "%^+_nouser")
+                std::cout << std::setw(20) << std::left << acct
+                          << std::setw(20) << std::right << ""
+                          << std::setw(20) << std::right << shrs
+                          << std::setw(20) << std::right << usg
+                          << std::endl;
+            else
+                std::cout << std::setw(20) << std::left << acct
+                          << std::setw(20) << std::right << username
+                          << std::setw(20) << std::right << shrs
+                          << std::setw(20) << std::right << usg
+                          << std::endl;
+        }
+
+
+    }
+
+    return 0;
+}
+
+
+int main (int argc, char** argv)
+{
+    bool parsable = false;
+    std::string filepath;
+    int rc = 0;
+
+    for (int i = 1; i < argc; ++i) {
+        std::string arg = argv[i];
+        if (arg == "-h" || arg == "--help") {
+            show_usage ();
+            return rc;
+        } else if (arg == "-P")
+            parsable = true;
+    }
```

Comment:
Might be worth adding an `else` to print an error on an unknown argument.

Reply:
Ah, yeah, thank you for pointing this out. I pushed up a fix that should print the usage for `flux shares` and then sets `rc` to -1 if a bad argument is passed in.